thriftrw/spec/struct.pyx

```diff
@@ -134,6 +135,7 @@ cdef class StructTypeSpec(TypeSpec):
         if not self.linked:
             self.linked = True
             self.fields = [field.link(scope) for field in self.fields]
+            self.hashable = all([f.hashable for f in self.fields])
```

Comment:
don't need the `[` around the `all` I think.

```suggestion
            self.hashable = all(f.hashable for f in self.fields)
```

Reply:
I think we do, `cython` complained because apparently it doesn't support closures.

Reply:
It's not a closure. It's an iterator.

    x = (f.hashable for f in self.fields)
    
Is equivalent to,

    def hashables(self):
        for f in self.fields:
            yield f.hashable
    
    x = hashables(self)

The parens are required only to appease the parser. This wouldn't work:

    x = f.hashable for f in self.fields
    
But when passed into a function as the only argument, it isn't necessary.

    foo((f.hashable for f in self.fields))  # ==
    foo(f.hashable for f in self.fields)
    
---

That said, this is fine if Cython doesn't like iterators being passed like
that.

Reply:
Cython calls it a closure :)

> closures inside cpdef functions not yet supported